fe/fe-core/src/main/java/com/starrocks/mysql/privilege/Auth.java

```diff
@@ -620,6 +620,11 @@ public void dropUser(DropUserStmt stmt) throws DdlException {
             // Allow dropping `superuser@%` when doing `DROP CLUSTER`, but not for `DROP USER`.
             throw new DdlException(String.format("User `%s`@`%s` is not allowed to be dropped.", user, host));
         }
+
+        if (!doesUserExist(stmt.getUserIdentity())) {
```

Comment:
Why not do this check in `dropUserInternal`
Some concurrent clients will drop this user after this clause.
It's better to make checking and removing as one atomic operation